commons/src/main/java/fr/free/nrw/commons/upload/MwVolleyApi.java

```diff
@@ -18,23 +19,75 @@
 
 
 import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
 
 public class MwVolleyApi {
 
     private static RequestQueue REQUEST_QUEUE;
     private static final Gson GSON = new GsonBuilder().create();
     private Context context;
+    private static String coordsLog;
+
+    protected static HashSet<String> categorySet;
+
+    //To check later on whether any nearby categories were found
+    public static boolean gpsCatExists;
 
     public MwVolleyApi(Context context) {
         this.context = context;
+        categorySet = new HashSet<String>();
+    }
+
+    //To get the list of categories for display
+    public static ArrayList<String> getGpsCat() {
```

Comment:
Why not return just a `List`? Are methods from an ArrayList implementation needed?